controllers/progressivesync_controller_test.go

```diff
@@ -82,10 +85,34 @@ var _ = Describe("ProgressiveRollout Controller", func() {
 	)
 
 	BeforeEach(func() {
+		ctx, cancel = context.WithCancel(context.Background())
+
 		namespace, ns = createRandomNamespace()
+
+		k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+			Scheme:    scheme.Scheme,
+			Namespace: namespace,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		mockAcdClient := mocks.ArgoCDAppClientStub{}
+		reconciler = &ProgressiveSyncReconciler{
+			Client:          k8sManager.GetClient(),
+			Log:             ctrl.Log.WithName("controllers").WithName("progressivesync"),
+			ArgoCDAppClient: &mockAcdClient,
+		}
+		err = reconciler.SetupWithManager(k8sManager)
+		Expect(err).ToNot(HaveOccurred())
+
+		go func() {
+			defer GinkgoRecover()
+			err = k8sManager.Start(ctx)
```

Comment:
`k8sManager.Start` takes a context, so we pass the context we created with the cancel option and then send the signal to stop the manager with `defer cancel()` in `AfterEach`